modules/library/metadata/src/main/java/org/geotools/util/Utilities.java

```diff
@@ -755,4 +760,12 @@ public static void ensureArgumentNonNull(final String name, final Object object)
                 Function.identity(),
                 (first, second) -> second);
     }
+
+    /**
+     * @param fileName
+     * @return
+     */
+    public static void isZipSlipVulnerable(String fileName) throws IOException {
```

Comment:
Naming is a bit off... this method name makes it sound like something that would return a boolean, instead it throws an exception. Maybe "assertNotZipSlipVulnerable" or "checkZipSlip" instead?